hoot-core/src/main/cpp/hoot/core/io/HootNetworkRequest.cpp

```diff
@@ -150,19 +156,48 @@ bool HootNetworkRequest::_networkRequest(const QUrl& url, int timeout,
     return false;
     break;
   }
-  //  Wait for finished signal from reply object
-  _blockOnReply(reply, timeout);
-  //  Get the status and content of the reply if available
-  _status = _getHttpResponseCode(reply);
-  //  According to the documention this shouldn't ever happen
-  if (reply == NULL)
+  //  Start the timer
+  timeoutTimer.start(timeout * 1000);
+  //  Connect the timeout lambda
+  QObject::connect(&timeoutTimer, &QTimer::timeout, [&]()
+  {
+    //  Stop the timer first so the lambda isn't called again
+    timeoutTimer.stop();
+    //  Set the timed out flag before aborting the reply
+    _timedOut = true;
+    //  Abort will trigger QNetworkAccessManager::finished()
+    reply->abort();
+    //  Set the status to negative TimeoutError, negative so that it doesn't
+    //  collide with HTTP status numbers
+    _status = -1 * (int)QNetworkReply::TimeoutError;
+    _error = QString("HTTP Request to %1 timed-out.").arg(tempUrl.toString(QUrl::RemoveUserInfo));
+    //  Exit with a non-zero status
+    loop.exit(_status);
+  });
+  //  Connect the finished lambda
+//  QObject::connect(pNAM.get(), &QNetworkAccessManager::finished, [&]()
+  QObject::connect(reply, &QNetworkReply::finished, [&]()
   {
-    throw HootException(QString("Network request error: GET/POST/PUT request failed to create reply object."));
+    //  The finished signal is emitted when the reply is aborted in the
+    //  timeout lambda, so don't do anything
+    if (!_timedOut)
+    {
+      //  Stop the timer
+      timeoutTimer.stop();
+      _timedOut = false;
+      //  Exit with a zero status
+      loop.exit();
+    }
+  });
```

Comment:
The finished lambda stops the timer and exists to indicate that the read completed without timing-out.  The interesting thing here is that `finished` is triggered when a timeout occurs because of the call to `reply->abort()` above.  This requires that the `_timedOut` flag is set to `true` before aborting the read and skipping all of the other `finished` code.